crates/wasi-common/src/string_array_writer.rs

```diff
@@ -0,0 +1,53 @@
+use crate::{Error, Result};
+use std::convert::TryInto;
+use std::ffi::CString;
+use wiggle::GuestPtr;
+
+pub trait StringArrayWriter {
```

Comment:
How come this is a trait? Could this perhaps be a free-function in a utils-like module of some kind?

Reply:
My thought was we would eventually support different representations than `Vec<CString>` here as part of modularization. It seemed like a reasonable way to hang three methods onto some existing struct members. But maybe instead of `WasiCtx { args: Vec<CString>, env: Vec<CString> }` we could make this `struct StringArray(Vec<String>)` that just has these three pub methods?